### DISCUSSION TITLE
Further to Issue #56 - Modified AGPL license

### DISCUSSION THREAD

**Discussion by 3brover005** [docusealco/docuseal#70] [General]
Hi Alex, you closed the issue without truly elaborating. AGPL already forces companies to open source their modifications, so your additional terms added to the AGPL license are only focused on restricting the freedoms users have. Although you have every right to do so, I am worried about the false advertising that this is Open Source when in fact it goes against the principles of free software.

**Comment by 3brover005** [comment]
Just to elaborate further: AGPL explicitly states that only permissive modifications can be added, not restrictive. Please consider re-instating the standard AGPL license in order to keep the application covered by the principles of AGPL

**Reply by 3brover005** [reply]
Please acquaint yourself with the following case: https://sfconservancy.org/news/2023/feb/09/kuhn-neo4j-purethink-expert-report/

Neo4j added restrictive clauses, got sued and lost.

**Reply by AlexBTurchyn** [reply]
@3brover005 Neo4j Common Clause and "further restrictions" removal is a completely different case.
Common Clause from Neo4j is considered as "further restrictions" since it's not a subject of "additional terms" listed in Section 7.
At DocuSeal we don't have any "further rescription", there are "additional terms" provided in accordance to the Section 7(b) - author attribution.

>All other non-permissive additional terms are considered "further restrictions"

But the terms I added are not "all other non-permissive additional terms" they are the part of the 7(b) allowed additional terms listed above so they do not contradict with AGPL (that's the main difference with Neo4j case you mentioned).

Many similar AGPL projects have additional terms according to the section 7(b):
https://github.com/ONLYOFFICE/core/blob/master/DjVuFile/DjVu.cpp#L22
https://github.com/salesagility/SuiteCRM/blob/master/modules/Users/Logout.php#L37

Hope this makes sense but please let me know if I'm missing something.

I'm still thinking whether it's worth to add those additional terms or not because I like the simplicity of having a basic AGPLv3 (even though, I believe those additional terms comply with AGPL)

**Reply by 3brover005** [reply]
IANAL, but whenever AGPL is modified, it becomes vulnerable to legal interpretation. I am afraid I do not know what is the best way for you to protect the brand and logo of the app unfortunately..🤷‍♂️

**Comment by AlexBTurchyn** [comment]
@3brover005, docuseal is a commerial open source software, silimar to metabase, posthog, cal.com, HexaPDF and many other similar project with this business model. There is even a pricing page https://www.docuseal.co/pricing on our website. I think DocuSeal is not different from others - the project and users of the project will only benefit this COSS model since it will help me to fund the development of the OSS Docuseal and so to make it better to regular users that just install it on docker as is (usually only companies that want to copy/reuse OSS in their commercial software don't like AGPL).

**Reply by 3brover005** [reply]
That is perfectly good, you are allowed to do this, but you need to protect the app from the big companies with loads of $$$ for lawyers. I am not saying discontinue the current model, I am saying that you are violating AGPL, thus it wouldn't stand in the court of law. This opens your business and this project to risk from companies and individuals that are well versed in corporate and copyright laws.

**Comment by nanoscopic** [comment]
I don't understand this discussion. The claim seems to be that the project has modified the AGPL license but I don't see where.

@3brover005 Could you please share exactly which bits are added compared to the normal AGPL license to clarify what you are talking about?

The AGPL license requires changes to be open sourced if the modified version is being made available to people outside the company of the modifiers on the public internet.

I'm not clear though how Docuseal Co can have a private fork with additional features if they ever accepted open source contributions under the AGPL without a CLA. The moment such contributions were made it would mean Docuseal Co themselves are forbidden from having a private fork that is offered as a paid thing. The claim seems to be that the additional features are "modules", but that is clearly not the case from what I see in the code, and "modules" doesn't let you avoid the terms of the AGPL.

@docusealco Could you please explain how you are able to have a private fork legally without using a CLA for open source contributions to the project?

**Reply by omohokcoj** [reply]
@nanoscopic the only contributors to the AGPLv3 open-source code are DocuSeal LLC employees/contractors - the codebase doesn't have external open source contributions.

**Reply by nanoscopic** [reply]
That certainly works as a solution. Thank you for the explanation of how this is possible.
